include/nomp-log.h

```diff
@@ -47,8 +47,21 @@ int nomp_log_set_verbose(const int verbose);
 int nomp_log_(const char *desc, int logno, nomp_log_type type,
               const char *fname, unsigned line_no, ...);
 
-#define nomp_log(logno, type, desc, ...)                                       \
-  nomp_log_(desc, logno, type, __FILE__, __LINE__, ##__VA_ARGS__)
+#define nomp_num_helper(_1, _2, _3, _4, _5, _6, _7, _8, N, ...) N
+#define nomp_num(...) nomp_num_helper(__VA_ARGS__, 2, 2, 2, 2, 2, 2, 2, 1, 0)
+
+#define nomp_first_helper(first, ...) first
+#define nomp_first(...) nomp_first_helper(__VA_ARGS__, throwaway)
+
+#define nomp_rest_helper_2(first, ...) , __VA_ARGS__
+#define nomp_rest_helper_1(first)
+#define nomp_rest_helper_(num, ...) nomp_rest_helper_##num(__VA_ARGS__)
+#define nomp_rest_helper(num, ...) nomp_rest_helper_(num, __VA_ARGS__)
+#define nomp_rest(...) nomp_rest_helper(nomp_num(__VA_ARGS__), __VA_ARGS__)
+
+#define nomp_log(logno, type, ...)                                             \
```

Comment:
Shall we make these macros upper case as well similar to the test macros?

Reply:
I like the names used in the subtest macros (for example `IMPL` instead of `HELPER_`, IMPL_WITH_1` instead of `helper_1`, etc.). Can we rename these similarly as well?

Reply:
Probably `_CASE` is better than `_num` in both `nomp_num` and `TEST_ARGS_NUM`.